lib/Conversion/HWToBTOR2/HWToBTOR2.cpp

```diff
@@ -880,6 +880,12 @@ struct ConvertHWToBTOR2Pass
     genState(reg, w, regName);
 
     if (init) {
+      if (init && (isa<BlockArgument>(init) ||
+                   !isa<seq::InitialOp>(init.getDefiningOp()))) {
```

Comment:
any reason why you're checking init twice here?

Reply:
Oops! Just a copy-paste error, thanks for the catch!